lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -2223,25 +2228,21 @@ assignChangeAddressesAndUpdateDb ctx wid generateChange selection =
         (selectionUpdated, stateUpdated) =
             assignChangeAddresses generateChange selection s
 
-assignChangeAddressesWithoutDbUpdate
-    :: forall ctx s k.
-        ( GenChange s
-        , HasDBLayer IO s k ctx
-        )
-    => ctx
+-- | For use with the returned updated 's' of 'balanceTransaction'.
+writeChangeAddressStateToDb
+    :: AddressBookIso s
+    => DBLayer IO s k
     -> WalletId
-    -> ArgGenChange s
-    -> Selection
-    -> ExceptT ErrConstructTx IO (SelectionOf TxOut)
-assignChangeAddressesWithoutDbUpdate ctx wid generateChange selection =
-    db & \DBLayer{..} -> mapExceptT atomically $ do
-        cp <- withExceptT ErrConstructTxNoSuchWallet $
-            withNoSuchWallet wid $ readCheckpoint wid
-        let (selectionUpdated, _) =
-                assignChangeAddresses generateChange selection (getState cp)
-        pure selectionUpdated
+    -> s
+    -> IO ()
+writeChangeAddressStateToDb DBLayer{..} wid s = throwInIO $
+    atomically $ modifyDBMaybe walletsDB $
+        adjustNoSuchWallet wid id $ \_wallet ->
```

Comment:
Ignoring the `_wallet` feels a bit off, but
1. I think we should be using `atomicallyWithHandler` in the server handler anyway
2. I'm not sure: did the DB's `atomically` work as expected before this change? If yes, what's the point of `atomicallyWithHandler`?